cvmfs/ingestion/tube.h

```diff
@@ -65,17 +66,34 @@ class Tube : SingleCopy {
    * Push an item to the back of the queue.  Block if queue is currently full.
    */
   Link *Enqueue(ItemT *item) {
-    assert(item != NULL);
     MutexLockGuard lock_guard(&lock_);
     while (size_ == limit_)
       pthread_cond_wait(&cond_capacious_, &lock_);
 
     Link *link = new Link(item);
-    link->next_ = tail_;
-    link->prev_ = tail_->prev_;
-    tail_->prev_->next_ = link;
-    tail_->prev_ = link;
-    tail_ = link;
+    link->next_ = head_->next_;
```

Comment:
Isn't this putting the link to position two, instead of at the back of the list?

Reply:
I don't think so. head_ is sentinel, head_->prev is the first element, head_->next is the last element.